From a383f3ed37ed74bcb325e6b908cd9cf12f2baa88 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Tue, 6 Oct 2020 12:58:48 -0700
Subject: [PATCH 16/20] crypto: xor - Remove unused variable count in
 do_xor_speed

Clang warns:

crypto/xor.c:101:4: warning: variable 'count' is uninitialized when used
here [-Wuninitialized]
                        count++;
                        ^~~~~
crypto/xor.c:86:17: note: initialize the variable 'count' to silence
this warning
        int i, j, count;
                       ^
                        = 0
1 warning generated.

After the refactoring to use ktime that happened in this function, count
is only assigned, never read. Just remove the variable to get rid of the
warning.

Fixes: c055e3eae0f1 ("crypto: xor - use ktime for template benchmarking")
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
Link: https://github.com/ClangBuiltLinux/linux/issues/1171
Link: https://lore.kernel.org/r/20201006195848.707504-1-natechancellor@gmail.com
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 crypto/xor.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/crypto/xor.c b/crypto/xor.c
index a0badbc03577..eacbf4f93990 100644
--- a/crypto/xor.c
+++ b/crypto/xor.c
@@ -83,7 +83,7 @@ static void __init
 do_xor_speed(struct xor_block_template *tmpl, void *b1, void *b2)
 {
 	int speed;
-	int i, j, count;
+	int i, j;
 	ktime_t min, start, diff;
 
 	tmpl->next = template_list;
@@ -98,8 +98,6 @@ do_xor_speed(struct xor_block_template *tmpl, void *b1, void *b2)
 			mb(); /* prevent loop optimzation */
 			tmpl->do_2(BENCH_SIZE, b1, b2);
 			mb();
-			count++;
-			mb();
 		}
 		diff = ktime_sub(ktime_get(), start);
 		if (diff < min)
-- 
2.29.0.rc0

